lib/pengine/container.c

```diff
@@ -0,0 +1,800 @@
+/*
+ * Copyright (C) 2004 Andrew Beekhof <andrew@beekhof.net>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include <crm_internal.h>
+
+#include <crm/pengine/rules.h>
+#include <crm/pengine/status.h>
+#include <crm/pengine/internal.h>
+#include <unpack.h>
+#include <crm/msg_xml.h>
+
+#define VARIANT_CONTAINER 1
+#include "./variant.h"
+
+void tuple_free(container_grouping_t *tuple);
+
+static char *
+next_ip(const char *last_ip)
+{
+    int oct1 = 0;
+    int oct2 = 0;
+    int oct3 = 0;
+    int oct4 = 0;
+
+    int rc = sscanf(last_ip, "%d.%d.%d.%d", &oct1, &oct2, &oct3, &oct4);
+    if (rc != 4) {
+        return NULL;
+
+    } else if(oct4 > 255) {
+        return NULL;
+    }
+
+    return crm_strdup_printf("%d.%d.%d.%d", oct1, oct2, oct3, oct4+1);
```

Comment:
You don't want to use 255, and you're adding 1, so the upper bound should be (oct4 >= 254). If we want to support more than 254 container instances, we should roll over oct3 instead.